eureka-services/src/main/java/edu/emory/cci/aiw/cvrg/eureka/services/resource/AppPropertyResource.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package edu.emory.cci.aiw.cvrg.eureka.services.resource;
+
+import com.google.inject.Inject;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import edu.emory.cci.aiw.cvrg.eureka.common.props.AppPropertiesMode;
+import edu.emory.cci.aiw.cvrg.eureka.common.props.AppPropertiesRegistration;
+import edu.emory.cci.aiw.cvrg.eureka.common.props.AppPropertiesLinks;
+import java.util.ArrayList;
+import java.util.List;
+/**
+ *
+ * @author Miao Ai
+ */
+
+@Path("/appproperties")
+@Produces(MediaType.APPLICATION_JSON)
+public class AppPropertyResource {
+    
+	private final AppPropertiesMode appPropertiesMode;
+	private final AppPropertiesLinks appPropertiesLinks;
+	private final AppPropertiesRegistration appPropertiesRegistration;
+        
+	@Inject
+	public AppPropertyResource( AppPropertiesMode inAppPropertiesMode, 
```

Comment:
Inject ServiceProperties here instead, and create each of the data transfer objects in the getter, passing the ServiceProperties instance into each data transfer object's 1-arg constructor.